state/pruner.go

```diff
@@ -160,7 +176,10 @@ func (p *Pruner) SetCompanionRetainHeight(height int64) error {
 	if currentCompanionRetainHeight > height || (!noAppRetainHeight && currentAppRetainHeight > height) {
 		return ErrPrunerCannotLowerRetainHeight
 	}
-	return p.stateStore.SaveCompanionBlockRetainHeight(height)
+	if err = p.stateStore.SaveCompanionBlockRetainHeight(height); err == nil {
+		p.metrics.PruningServiceBlockRetainHeight.Set(float64(height))
+	}
+	return err
```

Comment:
The typical "early return" pattern's probably the better/clearer one here:

```suggestion
	if err := p.stateStore.SaveCompanionBlockRetainHeight(height); err != nil {
		return err
	}
	p.metrics.PruningServiceBlockRetainHeight.Set(float64(height))
	return nil
```

Reply:
Sure, although for this particular function it seemed readable enough ,but will refactor.

Reply:
With this  change the linter complains again about code being to similar.